src/main/java/bio/terra/service/duos/DuosService.java

```diff
@@ -148,14 +153,108 @@ private void updateLastSynced(DuosFirecloudGroupModel firecloudGroup, Instant la
     try {
       duosDao.updateFirecloudGroupLastSyncedDate(id, lastSyncedDate);
     } catch (CannotSerializeTransactionException ex) {
-      String msg =
-          "Members of Firecloud group %s were updated, "
-              + "but could not update DUOS Firecloud Group record %s with last_synced_date %s"
-                  .formatted(firecloudGroup.getFirecloudGroupName(), id, lastSyncedDate);
+      String message =
+          firecloudGroup.getFirecloudGroupEmail()
+              + " members were updated, "
+              + "but an error occurred when updating its database record with last_synced_date "
+              + lastSyncedDate;
       String errorDetail =
           "This indicates that another process was updating the database row at the same time";
-      throw new DuosFirecloudGroupUpdateConflictException(msg, ex, List.of(errorDetail));
+      throw new DuosFirecloudGroupUpdateConflictException(message, ex, List.of(errorDetail));
+    }
+  }
+
+  /**
+   * @param id of the DUOS Firecloud Group record if it was successfully synced, null otherwise
+   * @param error encountered if the DUOS Firecloud Group failed to sync, null otherwise
+   */
+  record SyncResult(UUID id, ErrorModel error) {}
+
+  /**
+   * For all DUOS dataset IDs registered in TDR, force a sync of their Firecloud group members. A
+   * failure to sync any Firecloud group(s) should not block attempts to sync the rest.
+   *
+   * @return DUOS Firecloud groups whose members were successfully synced, and any errors occurred
+   *     which may have interfered with syncing
+   */
+  public DuosFirecloudGroupsSyncResponse syncDuosDatasetsAuthorizedUsers() {
+    List<ErrorModel> errors = new ArrayList<>();
```

Comment:
Now that you don't update it in your threads you no longer need to declare `errors` here, you can declare it below on line 192.

Reply:
Right!  The possibility of modifying the list again later made things a touch more involved.  But I reorganized the code so that collecting all of the errors happens together, I find it easier to follow.